### PR TITLE
DCP 549: Overwrite project metadata on spreadsheet upload

### PR DESCRIPTION
**Pull request by aaclan-ebi**
ebi-ait/dcp-ingest-central#549

### CODE REVIEW

File: src/app/shared/components/upload/upload.component.ts [hunk 1]
```diff
@@ -67,4 +76,9 @@ export class UploadComponent {
       this.alertService.error('No file chosen!', 'Please choose a spreadsheet to upload.', false, true);
     }
   }
+
+  onValueChanged($event: string) {
+    console.log('radio value changed')
```
Comment (amnonkhen) [inline src/app/shared/components/upload/upload.component.ts:81]: This should not be in the prod version.

File: src/app/project/project.component.ts [hunk 2]
```diff
@@ -199,10 +199,7 @@ export class ProjectComponent implements OnInit {
   }
 
   canSubmit(project: Project) {
-    return this.userIsWrangler &&
-      !project.hasOpenSubmission &&
-      project.validationState.toUpperCase().replace(' ', "_") !== SUBMISSION_STATES.GraphValid &&
-      !(project.validationErrors && project.validationErrors.length > 0);
+    return this.userIsWrangler && !project.hasOpenSubmission;
```
Comment (jacobwindsor) [inline src/app/project/project.component.ts:202]: Why remove all of the logic? It's quite nice not to submit it no?

File: src/app/shared/services/broker.service.spec.ts [hunk 3]
```diff
@@ -3,12 +3,13 @@ import {HttpClientTestingModule, HttpTestingController} from '@angular/common/ht
...
 import {BrokerService} from './broker.service';
 import {TimeoutError} from 'rxjs';
+import {environment} from "@environments/environment";
 
 describe('Broker Service', () => {
   let service: BrokerService;
   let httpTestingController: HttpTestingController;
   let httpClient: HttpClient;
-  const api_url = 'http://localhost:5000';
+  const api_url = environment.BROKER_API_URL;
```
Comment (jacobwindsor) [inline src/app/shared/services/broker.service.spec.ts:12]: Thanks!

**Review by jacobwindsor — APPROVED** [review]

File: src/app/project/project.component.ts (hunk 2 above)
Comment (aaclan-ebi) [reply src/app/project/project.component.ts:202]: This is per @gabsie's request. We now want to allow users to update the project upon spreadsheet upload. We want to let users upload a submission even if the project is invalid.

File: src/app/shared/components/upload/upload.component.html [hunk 4]
```diff
@@ -1,14 +1,25 @@
...
       <br/>
       <p>
-        <input type="file" #fileInput placeholder="Upload file..." />
+        <input type="file" #fileInput placeholder="Upload file..."/>
       </p>
 
+      <ng-container *ngIf="projectUuid && !isUpdate">
+        <mat-radio-group [(ngModel)]="updateProject">
+          <mat-radio-button [value]="false" [checked]="!updateProject">
+            <span>Do not update the project metadata. This will ignore project worksheet in the spreadsheet.</span>
```
Comment (ke4) [inline src/app/shared/components/upload/upload.component.html:13]: What about displaying `Do not update the project metadata` and `Update the project metadata` with bold and add the next sentences from above without bold in new line?

**Review by ke4 — DISMISSED** [review]
I just added 1 tiny suggestion as a comment, but otherwise it looks fine.

**Comment by aaclan-ebi** [comment]
Screenshot 2022-01-03 at 01 04 24

File: src/app/shared/components/upload/upload.component.html (hunk 4 above)
Comment (aaclan-ebi) [reply src/app/shared/components/upload/upload.component.html:13]: @gabsie what do you think?

**Comment by ke4** [comment]
> Screenshot 2022-01-03 at 01 04 24

Yep. I got that. I just thought that the 1 sentence with bold and the 2nd in new line would be a nicer design, at least for me. ;-)

**Review by ke4 — APPROVED** [review]

**Comment by aaclan-ebi** [comment]
Thanks @ke4 ! I agree it looks better.

Screenshot 2022-01-04 at 11 07 51

**Comment by aaclan-ebi** [comment]
Got an approval from @ke4 and @jacobwindsor last 23rd Dec 2021. Merging this now!

File: src/app/shared/components/upload/upload.component.html (hunk 4 above)
Comment (aaclan-ebi) [reply src/app/shared/components/upload/upload.component.html:13]: nvm, gonna merge and deploy this pr soon to dev and staging! Pls add any further comments on the ticket https://github.com/ebi-ait/dcp-ingest-central/issues/549#issuecomment-1004718994

**Pull request merged**